src/lib/Components/Bidding/Screens/BillingAddress.tsx

```diff
@@ -1,344 +1,244 @@
+              textContentType="addressCity"
+              returnKeyType="next"
+              enableClearButton
+              error={checkFieldError("city")}
+              value={address.city}
+              onChangeText={(e) => handleInputTextChange("city", e)}
+              testID="input-city"
+            />
+
+            <Input
+              ref={stateRef}
+              title="State, Province, or Region"
+              placeholder="Add state, province, or region"
+              textContentType="addressState"
+              returnKeyType="next"
+              enableClearButton
+              error={checkFieldError("state")}
+              value={address.state}
+              onChangeText={(e) => handleInputTextChange("state", e)}
+              testID="input-state-province-region"
+            />
+
+            <Input
+              ref={postalCodeRef}
+              title="Postal code"
+              placeholder="Add your postal code"
+              textContentType="postalCode"
+              returnKeyType="next"
+              enableClearButton
+              error={checkFieldError("postalCode")}
+              value={address.postalCode}
+              onChangeText={(e) => handleInputTextChange("postalCode", e)}
+              testID="input-post-code"
+            />
+
+            <Input
+              ref={phoneRef}
+              title="Phone"
+              placeholder="Add your phone number"
+              keyboardType="phone-pad"
+              textContentType="telephoneNumber"
+              returnKeyType="next"
+              enableClearButton
+              error={checkFieldError("phoneNumber")}
+              value={address.phoneNumber}
+              onChangeText={(e) => handleInputTextChange("phoneNumber", e)}
+              testID="input-phone"
+            />
+
+            <ScreenDimensionsContext.Consumer>
+              {({ height }) => {
+                const countryErr = checkFieldError("country")
+
+                return (
                   <Flex mb={4}>
                     <CountrySelect
+                      ref={countryRef}
                       maxModalHeight={height * 0.95}
-                      onSelectValue={(value) => {
-                        this.onCountrySelected({
-                          shortName: value,
-                          longName: COUNTRY_SELECT_OPTIONS.find((opt) => opt.value === value)!.label,
-                        } as Country)
-                      }}
-                      value={this.state.values.country?.shortName}
-                      hasError={!!errorForCountry}
+                      onSelectValue={(value: string) => handleCountrySelection(value)}
+                      value={address.country.shortName}
+                      hasError={!!countryErr}
                     />
-                    {!!errorForCountry && (
-                      <Sans size="2" color="red100">
-                        {errorForCountry}
+                    {!!countryErr && (
```

Comment:
I looked in the code base and I think naming errors `error` instead of `err` is more commen.
```suggestion
                    {!!countryError && (
```